lessons/5.mdx

```diff
@@ -0,0 +1,1362 @@
+Rainbowkit.
+- `import { getDefaultWallets, RainbowKitProvider, } from '@rainbow-me/rainbowkit';` returns
+the default wallet provider option and a RainbowKit Provider to "wrap" around our app
+so that we can use its features throughout the app's pages and components.
+- `import { chain, configureChains, createClient, WagmiConfig, } from 'wagmi';` provides
+a way to configure the WagmiConfig wrapper for our app based on the chains and providers
+of our choice along with some other customisations.
+- `import { alchemyProvider } from 'wagmi/providers/alchemy';` and 
+`import { publicProvider } from 'wagmi/providers/public';` give us the provider configs
+for the providers we will be using. In our app we are using an `alchemyProvider` along
+with a fall back `publicProvider`. 
+- `useEffect` and `useState` are react hooks that help us perform side effects and
+capture the state, respectively.
+
+After importing we create the configurations we need and create a new instance of 
+`wagmiClient` using them.
+
+```jsx
+// import statements code...
+
+const { chains, provider } = configureChains(
+  [chain.goerli],
+  [
+    alchemyProvider({ apiKey: process.env.ALCHEMY_ID }),
+    publicProvider()
+  ]
+);
+
+const { connectors } = getDefaultWallets({
+  appName: 'My RainbowKit App',
+  chains
+});
+
+const wagmiClient = createClient({
+  autoConnect: true,
+  connectors,
+  provider
+})
+
+// function MyApp code...
+```
+
+For this application we will continue using the `goerli` testnet. Other chains can
+be added simply using the following syntax: `chain.chain_name`.
+For custom providers, like the `alchemyProvider` we can pass in our private
+`apiKey` as well.
+
+Now we can wrap our application in the `RainbowKitProvider` and `WagmiConfig`
+so that have access to their features throughout our application.
+
+```jsx
+// import statements and configs code...
+
+function MyApp({ Component, pageProps }) {
+  return (
+    <WagmiConfig client={wagmiClient}>
+      <RainbowKitProvider chains={chains}>
+        <Component {...pageProps} />
+      </RainbowKitProvider >
+    </WagmiConfig >
+  );
+};
+
+export default MyApp
+```
+
+Now that we have our application setup, we can edit the Index Page.
+<br/>
+
+---
+
+### The Index Page
+
+The `index.js` page in the `/frontend/pages` directory should look like this 
+after deleting all the children in the `<main>` tag and deleting the `<footer>`.
+
+```jsx
+import Head from 'next/head'
+import Image from 'next/image'
+import styles from '../styles/Home.module.css'
+
+export default function Home() {
+  return (
+    <div className={styles.container}>
+      <Head>
+        <title>Create Next App</title>
+        <meta name="description" content="Generated by create next app" />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <main className={styles.main}>
+      </main>
+    </div>
+  )
+}
+```
+
+Before we dive into the code, let's copy and paste some styling from the 
```

Comment:
Before we dive into the ........ code - It would  be nice if you said which type of code here e.g. the **next.js** code. It would give the learner a sense that they are using different layers to build up that **full stack** you mention at the start.